esi_leap/api/controllers/v1/node.py

```diff
@@ -47,6 +56,16 @@ def get_all(self):
         nodes = ironic.get_node_list(context)
 
         node_collection = NodeCollection()
-        node_collection.nodes = [Node(name=n.name) for n in nodes]
+
+        for node in nodes:
+            offers = offer_obj.Offer.get_all({"resource_uuid": node.uuid},
+                                             context)
+            n = Node(name=node.name, uuid=node.uuid, owner=node.owner)
+            if offers:
+                n.offer_uuid = offers[0].uuid
+            if "lease_uuid" in node.properties:
+                n.lease_uuid = node.properties["lease_uuid"]
```

Comment:
This only shows the current lease; if you decide to only show the current offer, then this is enough; if you decide to show active and future offers, then you should do the same with the lease. Either way, adding in the lease's start/end time would be useful as well!